qiskit_machine_learning/algorithms/effective_dimension.py

```diff
@@ -0,0 +1,370 @@
+        num_params: int = 1,
+        num_inputs: int = 1,
+        callback: Optional[Callable[[str], None]] = None,
+        seed: int = 0,
+    ) -> None:
+
+        """
+        Args:
+            qnn: A Qiskit :class:`~qiskit_machine_learning.neural_networks.NeuralNetwork`,
+                with a specific dimension ``(num_weights)`` that will determine the shape of the
+                Fisher Information Matrix ``(num_inputs * num_params, num_weights, num_weights)``
+                used to compute the global effective dimension for a set of ``inputs``,
+                of shape ``(num_inputs, qnn_input_size)``,
+                and ``params``, of shape ``(num_params, num_weights)``.
+            params: An array of neural network parameters (weights), of shape
+                ``(num_params, num_weights)``.
+            inputs: An array of inputs to the neural network, of shape
+                ``(num_inputs, qnn_input_size)``.
+            num_params: If ``params`` is not provided, the algorithm will
+                randomly sample ``num_params`` parameter sets from a
+                uniform distribution. By default, ``num_params = 1``.
+            num_inputs:  If ``inputs`` is not provided, the algorithm will
+                randomly sample ``num_inputs`` input sets
+                from a normal distribution. By default, ``num_inputs = 1``.
+            callback: A callback function for the Monte Carlo sampling.
+        """
+
+        # Store arguments
+        self._params = None
+        self._inputs = None
+        self._seed = seed
+        self._num_params = num_params
+        self._num_inputs = num_inputs
+        self._model = qnn
+        self._callback = callback
+
+        # Define inputs and parameters
+        self.params = params  # type: ignore
+        # input setter uses self._model
+        self.inputs = inputs  # type: ignore
+
+    def num_weights(self) -> int:
+        """Returns the dimension of the model according to the definition
+        from [1]."""
+        return self._model.num_weights
+
+    @property
+    def seed(self) -> int:
+        """
+        Get seed.
+        """
+        return self._seed
+
+    @seed.setter
+    def seed(self, seed: int) -> None:
+        """
+        Set seed.
+
+        Args:
+            seed (int): seed to use.
+        """
+        self._seed = seed
+
+    @property
+    def params(self) -> np.ndarray:
+        """Returns network parameters."""
+        return self._params
+
+    @params.setter
+    def params(self, params: Optional[Union[List[float], np.ndarray, float]]) -> None:
+        """Sets network parameters."""
+        if params is not None:
+            self._params = np.asarray(params)
+            self._num_params = len(self._params)
+        else:
+            # random sampling from uniform distribution
+            np.random.seed(self._seed)
+            self._params = np.random.uniform(0, 1, size=(self._num_params, self._model.num_weights))
```

Comment:
As I commented before I think this should be using the algorithm_globals random generator as is done elsewhere. I.e. call `algorithm_globals.random.uniform(....)` which is a the random source, which can be seeded for reproducibility and can be used just the same way as above (its backed by a numpy Generator). The code above is using the legacy random function of numpy - if it is really essential to have instances of these on different generators, i.e. using a common core generator as is done elsewhere is not appropriate, then the above code should be changed from legacy. To me I do not see a good reason not to use the algorithm globals generator.